File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>s;
        s[0] = 1;
        int sum = 0;
        int count = 0;
        for(int i = 0; i < nums.size(); i++){
            sum += nums[i];
            if(s.find(sum-k) != s.end()){
                count += s[sum-k];
            }
            s[sum]++;
        }
        return count;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: list[int], k: int) -> int:
        mp = {0: 1} 
        sum1 = 0
        count = 0
        
        for i in nums:
            sum1 += i
            
            if (sum1 - k) in mp:
                count += mp[sum1 - k]

            mp[sum1] = mp.get(sum1, 0) + 1
            
        return count

        
